src/tests/testHttpHeaderTools.cc

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 1996-2023 The Squid Software Foundation and contributors
+ *
+ * Squid software is distributed under GPLv2+ license and includes
+ * contributions from numerous individuals and organizations.
+ * Please see the COPYING and CONTRIBUTORS files for details.
+ */
+
+#include "squid.h"
+#include "compat/cppunit.h"
+#include "HttpHeaderTools.h"
+#include "unitTestMain.h"
+
+#include <climits>
+#include <list>
+#include <map>
+#include <stdexcept>
+#include <string>
+#include <tuple>
+
+class TestHttpHeaderTools: public CPPUNIT_NS::TestFixture
+{
+    CPPUNIT_TEST_SUITE( TestHttpHeaderTools );
+    /* note the statement here and then the actual prototype below */
+    CPPUNIT_TEST( testHttpHeaderParseInt );
+    CPPUNIT_TEST( testHttpHeaderParseOffset );
```

Comment:
AFAICT, this PR adds unit tests for two functions:

* httpHeaderParseInt() -- essentially contains just two calls: atoi() and xisdigit()
* httpHeaderParseOffset() -- essentially contains just one call: strtoll()

This PR also proposes to use dozens of testHttpHeaderTools dependencies and three new stub files, some of them controversial and/or unpolished. For ease of reference, I will call all that stuff "baggage".

If testing the above two functions requires this baggage because of _other_ functions in src/HttpHeaderTools.cc, then please move the two tested functions into a file (probably somewhere inside src/http/ and Http namespace), and then remove nearly all the test dependencies and stubs. Let's focus on adding good unit tests -- a difficult and important task on its own!

Otherwise, please discuss why this baggage must be added in this PR, keeping in mind that some of the stuff in HttpHeaderTools.cc should _not_ be tested using unit tests _at all_ (but introduces heavy dependencies).

Reply:
Sound advice, I'll look to reduce the footprint of that file and make it smaller.
Note that eventually I plan to add more unit tests while waiting for other PRs to be cleared.
My objectives are:
* Use the wait time productively
* improve our unit tests coverage (duh)
* set the stage for test-driven rewriting of complex functions to more broadly support SBuf (See discussion about performance regressions in PR 1451